go.mod

```diff
@@ -1,10 +1,6 @@
 module github.com/grafana/mimir
 
-go 1.23.8
-
-// Please note that this directive is ignored when building with the Mimir build image,
-// that will always use its bundled toolchain.
-toolchain go1.24.2
+go 1.24.3
```

Comment:
Should this match the Go version in the build image?

And should we keep the `toolchain` line?

Reply:
> Should this match the Go version in the build image?

Ah yes, nicely spotted. 

> And should we keep the toolchain line?

I would prefer it, although `toolchain` is dropped by `go mod tidy` automatically when upgrading.

Reply:
I tried adding it back, but `go mod tidy` strips it away consistently.

Reply:
Fixed.